### PR TITLE
Updates install CLI doc with more detail

### LINKED ISSUE #83: Improve notation download and installation instructions
The only installation experience that Notation currently has is for users to download the binary and manually copy to the appropriate folder on their machine. At a minimum, we need to provide the right instructions how to download, verify the binaries, and the correct places to copy the binaries for various operating systems. 

Starting a HackMD spec for that on https://hackmd.io/@toddysm/SJhUuI7Bi

Related information is available in https://github.com/notaryproject/notation/blob/main/specs/directory.md

@zr-msft to prepare an article

### PR DESCRIPTION
**Pull request by zr-msft**
Signed-off-by: Zach Rhoads <zachary.rhoads@microsoft.com>

fixes #83

Updates to CLI install instructions for platforms

### CODE REVIEW

**Comment by zr-msft** [comment]
@toddysm let me know what you think

**Comment by zr-msft** [comment]
@toddysm updated with section for verifying checksum with existing checksum file. Not sure if @yizha1 needs to change anything.

**Review by toddysm — APPROVED** [review]

**Comment by zr-msft** [comment]
@dtzar mind merging when you have a chance?

File: content/en/docs/installation/cli.md [hunk 1]
```diff
@@ -4,27 +4,89 @@ description: Install the notation CLI on Linux, macOS, and Windows
...
-Expand the archive file and add it to your path. For example, the following command expands the archive file directly to `~/bin`.
+* `notation_0.12.0-beta.1_darwin_amd64.tar.gz` for macOS running on Intel processors.
+* `notation_0.12.0-beta.1_darwin_arm64.tar.gz` for macOS running on Apple Silicon processors.
+* `notation_0.12.0-beta.1_linux_amd64.tar.gz` for Linux running on Intel processors.
+* `notation_0.12.0-beta.1_linux_arm64.tar.gz` for Linux running on ARM processors.
+* `notation_0.12.0-beta.1_windows_amd64.zip` for Windows running on Intel processors.
 
-```console
-tar xvzf notation.tar.gz -C ~/bin notation
+## Verify the `notation` CLI checksum
```
Comment (FeynmanZhou) [inline content/en/docs/installation/cli.md:19]: Is it necessary to showcase how to verify checksum in the installation guide? I think we can note this step is recommended but not a MUST.

File: content/en/docs/installation/cli.md [hunk 2]
```diff
@@ -4,27 +4,89 @@ description: Install the notation CLI on Linux, macOS, and Windows
 weight: 1
 ---
 
-## Download the `notation` CLI 
+## Download the `notation` CLI
 
-Download the latest CLI for your platform from the [Releases page](https://github.com/notaryproject/notation/releases).
+Navigate to the [Releases page](https://github.com/notaryproject/notation/releases) and select the latest release of the CLI. The notation CLI binaries for each platform are available under the *Assets* section in the `notation_<version>_<platform>.tar.gz` files.
```
Comment (FeynmanZhou) [inline content/en/docs/installation/cli.md:9]: From a user's perspective, it's inconvenient for me to just copy and paste to finish the installation step. I would suggest providing download & installation commands and consolidating them into different tabs.  See the [Minikube getting start page](https://minikube.sigs.k8s.io/docs/start/) as an example:

image

Comment (FeynmanZhou) [reply content/en/docs/installation/cli.md:9]: This format can be improved later after I confirmed with @toddysm . I will create an issue to track it.

**Review by FeynmanZhou — APPROVED** [review]

**Review by yizha1 — COMMENTED** [review]
LGTM

**Pull request merged**